linera-views/tests/operations_tests.rs

```diff
@@ -4,132 +4,199 @@
 use async_lock::{Mutex, RwLock};
 use linera_views::{
     batch::Batch,
-    common::{KeyIterable, KeyValueStoreClient},
+    common::{get_interval, KeyIterable, KeyValueIterable, KeyValueStoreClient},
     key_value_store_view::ViewContainer,
-    memory::MemoryContext,
-    test_utils::get_random_key_value_vec_prefix,
+    memory::{create_test_context, create_test_memory_client},
+    test_utils::{get_random_byte_vector, get_random_key_value_vec_prefix, get_small_key_space},
 };
-use rand::SeedableRng;
+use rand::{Rng, SeedableRng};
 use std::{
-    collections::{BTreeMap, HashMap},
+    collections::{BTreeMap, HashSet},
     sync::Arc,
 };
 
 #[cfg(feature = "rocksdb")]
-use linera_views::rocksdb::DB;
+use linera_views::rocksdb::create_rocksdb_test_client;
 
 #[cfg(feature = "aws")]
-use linera_views::{
-    dynamo_db::DynamoDbClient, lru_caching::TEST_CACHE_SIZE, test_utils::LocalStackTestContext,
-};
+use linera_views::test_utils::create_dynamodb_test_client;
 
+/// This test starts with a collection of key/values being inserted into the code
+/// which is then followed by a number of reading tests. The functionalities being
+/// tested are all the reading functionalities:
+/// * read_key_bytes
+/// * read_multi_key_bytes
+/// * find_keys_by_prefix / find_key_values_by_prefix
+/// * The ordering of keys returned by find_keys_by_prefix and find_key_values_by_prefix
 #[cfg(test)]
-async fn test_ordering_keys_key_value_vec<OP: KeyValueStoreClient + Sync>(
+async fn test_readings_vec<OP: KeyValueStoreClient + Sync>(
     key_value_operation: OP,
     key_value_vec: Vec<(Vec<u8>, Vec<u8>)>,
 ) {
     // We need a nontrivial key_prefix because dynamo requires a non-trivial prefix
-    let key_prefix = vec![0];
     let mut batch = Batch::new();
-    for key_value in key_value_vec {
-        batch.put_key_value_bytes(key_value.0, key_value.1);
+    let mut keys = Vec::new();
+    let mut set_keys = HashSet::new();
+    for key_value in &key_value_vec {
+        keys.push(key_value.0.clone());
+        set_keys.insert(key_value.0.clone());
+        batch.put_key_value_bytes(key_value.0.clone(), key_value.1.clone());
     }
     key_value_operation.write_batch(batch, &[]).await.unwrap();
-    let keys: Vec<Vec<u8>> = key_value_operation
-        .find_keys_by_prefix(&key_prefix)
-        .await
-        .unwrap()
-        .iterator()
-        .map(|x| {
-            let mut v = key_prefix.clone();
-            v.extend(x.unwrap());
-            v
-        })
-        .collect();
-    for i in 1..keys.len() {
-        let key1 = keys[i - 1].clone();
-        let key2 = keys[i].clone();
-        assert!(key1 < key2);
+    for key_prefix in keys
+        .iter()
+        .flat_map(|key| (1..key.len()).map(|u| &key[..=u]))
```

Comment:
```suggestion
        .flat_map(|key| (0..key.len()).map(|u| &key[..=u]))
```

I think…? Because we use `key[..=u]`; and `key[..=0]` is not empty.